edgedb-protocol/src/descriptors.rs

```diff
@@ -38,6 +41,49 @@ pub enum Descriptor {
     TypeAnnotation(TypeAnnotationDescriptor),
 }
 
+#[derive(Clone, PartialEq, Eq)]
+pub struct DescriptorUuid(Uuid);
+
+impl Debug for DescriptorUuid {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        match uuid_to_known_name(&self.0) {
+            Some(known_name) => write!(f, "{known_name}"),
+            None => write!(f, "{}", &self.0)
+        }
+    }
+}
+
+impl Deref for DescriptorUuid {
+    type Target = Uuid;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<Uuid> for DescriptorUuid {
+    fn from(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
+}
+
+impl FromStr for DescriptorUuid {
```

Comment:
Why you need FromStr? I'm not sure it's a good idea.

Reply:
It's a nice trait to have to keep the codec tests unchanged because they all use .parse() which needs FromStr, but would be easy enough (and still on a single line) to parse on the Uuid inside and then make them into a DescriptorUuid that way. Any preference? I'm okay with either way.